### ISSUE TITLE
Inconsistent use of parentheses for function/method references in documentation

### ISSUE THREAD

**Issue by jnqnfe** [rust-lang/rust#79679]
I'm noticing that, as titled, there are inconsistencies here and there within the `std` documentation (and core, alloc, etc) as to whether or not parentheses are used in references to methods and functions. Additionally there are even some method references that start with a fullstop.

I do not know what the policy is, but it appears (without thorough checks) to be so that the majority of references do _not_ use them. I'm not sure which way is best, but there should certainly be consistency.

I'm doing a little work to try to fix those that I can find and will submit a PR shortly.

I'm afraid that I can only afford to spend a little time right now removing parentheses from the handful of instances that use them. If a consensus were to develop that having them everywhere were more clear, that would have to be tackled separately later, by others if I cannot later find time for it.

**Comment by jnqnfe** [comment]
One difficulty in fixing the inconsistency is the problem of references that are subtly meant as code fragments.

For instance here the parentheses need to remain because it's a code fragment rather than just a reference.
```rust
/// This has the same effect as [`MaybeUninit::zeroed().assume_init()`][zeroed].
/// It is useful for FFI sometimes, but should generally be avoided.
```

What about this next one. We have no link, and it would be rather unclear to remove the parentheses, but it's a reference to a function rather than a fragment, so for consistency _should_ have them removed, but removing them would make it more difficult to understand.
```rust
/// This is usually the same as `size_of::<T>()`. However, when `T` *has* no
/// statically-known size, e.g., a slice [`[T]`][slice] or a [trait object],
/// then `size_of_val` can be used to get the dynamically-known size.
```

In this next case, removing the parentheses reduces clarity, but yet it's a reference so they must go.
```rust
/// `ManuallyDrop` robustly prevents double-free because we disable `v`'s destructor
/// before doing anything else. `mem::forget()` doesn't allow this because it consumes its
```

I think there's good grounds for arguing that having parentheses is better.

**Comment by jnqnfe** [comment]
And then we have this, where methods are being demonstrated in code fragments which are really meant as references, since `layout` is a mock variable that was not introduced in any prior example.
```rust
/// * The block must be allocated with the same alignment as [`layout.align()`], and
///
/// * The provided [`layout.size()`] must fall in the range `min ..= max`, where:
///   - `min` is the size of the layout most recently used to allocate the block, and
///   - `max` is the latest actual size returned from [`alloc`], [`grow`], or [`shrink`].
///
/// [`layout.align()`]: Layout::align
/// [`layout.size()`]: Layout::size
```

**Comment by camelid** [comment]
Personally, I don't think it really matters. Sometimes rustdoc requires parentheses to disambiguate, but other than that I think it should just be the stylistic choice of the author of the documentation. Just like how the rest of the documentation (e.g. wording) is based on the author's writing style.

**Comment by jnqnfe** [comment]
> Personally, I don't think it really matters. Sometimes rustdoc requires parentheses to disambiguate, but other than that I think it should just be the stylistic choice of the author of the documentation. Just like how the rest of the documentation (e.g. wording) is based on the author's writing style.

No it's not a big deal, but I'm not a fan of this concept of a mix of such styles and it's just... imperfect. Keeping a consistent writing style is asking way too much, but use of parentheses for functions/methods is a fairly reasonable thing to request consistency on imo. Additionally the lack of consistency and/or lack of parentheses does impact readability to some varying degree I feel, particularly when it fluctuates within a page or especially a single piece of documentation. Having parentheses gives a clear visual clue that distinguishes method/function references from other things. This is helpful surely to readers of all skill/experience levels, but particularly those with low/little I fear. Lacking parentheses and thus that visual clue can in places enhance or create uncertainty about the type of thing being referenced by broadening the possibilities somewhat. It can thus cause a distraction as the reader then needs to focus more and possibly hover/click the link to disambiguate. Lack of consistency, seeing some references with parentheses alongside ones without, surely can compound that potential uncertainty in an inexperienced reader. Maybe I'm making a big meal out nothing. I don't know, I'm not new/inexperienced myself and did not experience any particular difficulty; it's just that the docs are of excellent quality, and seeing this imperfection, I jumped onto it to try and get it ironed out.

As an update to the work I was doing, I got through "fixing" everything findable via the "()`]" pattern (to not use parentheses), but then there's a ton of references that are not links (only findable via "()`" rather than "()`]") that would also need fixing that I barely touched. Essentially it was all just a waste of effort. I underestimated how much work it would take to "fix" things with all-no-parentheses style for now (until some possible future where everything is fixed to have parentheses). Also, the more I did, the more strongly I felt that what I was accomplishing was just making the documentation worse in many places. So I think Ill just junk that work.

As above I feel that there's some value in adding parentheses in some places at least where they are missing, if not everywhere, but locating the places where they are missing is much more difficult than finding places where they are not missing, at least in terms of trying to search to save time rather than scrolling through every file which of course is very easy but very time consuming. I have too much else to do, and this isn't really worth spending much time on, so I don't know if I'll bother with any further work on it, at least for the time being.
